### PR TITLE
Disable deprecation warnings by default

### PR DESCRIPTION
**Pull request by hannes-steffenhagen-diffblue**
Disable deprecation warnings by default, to avoid masking other issues with them.

See discussion on https://github.com/diffblue/cbmc/pull/2037

### CODE REVIEW

**Review by chrisr-diffblue — APPROVED** [review]
Looks ok to me, but should wait for CI to run, of course.

**Comment by tautschnig** [comment]
I think it would be nice if @kroening could voice his opinion for he raised the concern in  the first place.

**Comment by hannes-steffenhagen-diffblue** [comment]
@kroening This re-enables the deprecated attribute for the conversion functions, but disables the warnings by default. This way people can enable the warnings during development to be warned about using deprecated functions but it doesn't show up in the logs when its not wanted. Is that in line with what you needed?

**Review by tautschnig — APPROVED** [review]

**Comment by tautschnig** [comment]
As the Travis build logs don't show any warnings it should be ok to merge this once CI  completes.

**Review by chrisr-diffblue — APPROVED** [review]
I think this is a sensible resolution for the moment. A separate PR/discussion can cover enabling the warnings on a single travis config if that is still desired.

**Pull request merged**
